packages/atomic/src/utils/props-utils.ts

```diff
@@ -38,31 +32,16 @@ export function MapProp(opts?: MapPropOptions) {
   };
 }
 
-export function ArrayProp(opts: ArrayPropOptions) {
-  const logWithDeprecation = (msg: string, ...other: unknown[]) =>
-    opts.deprecationWarning
-      ? console.warn(
-          `${msg} This will be enforced in the next major version`,
-          other
-        )
-      : console.error(msg, other);
-
+export function ArrayProp() {
   return (component: ComponentInterface, variableName: string) => {
     const {componentWillLoad} = component;
     const attributeWithBackets = camelToKebab(variableName);
 
-    if (!componentWillLoad) {
```

Comment:
Why was `componentWillLoad` required?

Reply:
i think its something that Stencil fixed at some point